### ISSUE TITLE
Question: any export gives 400, what can we do?

### ISSUE THREAD

**Issue by OlavAllema** [open-zaak/open-zaak#1102]
**General information**
Open Zaak version: 1.5

**Describe the question**
Export of any zaaktype or catalogus gives 400 Bad Request. 

**To Reproduce**
1. Go to a zaaktype
2. Click Export
3. Bad Request 400

**Expected behavior**
In the API specificaties we don't see an API for this.
In Portainer we see that there is this POST /admin/catalogi/zaaktype/12/change/?_changelist_filters=catalogus__id__exact%3D3
We can't see any errors on out side. 
Can you provide us some info (or error) on this?

Or is this something that could cause the problem?
<img width="1128" alt="image" src="https://user-images.githubusercontent.com/75848616/154493818-7689177d-e292-4d43-8178-c842a2d5e68e.png">

**Comment by sergei-maertens** [comment]
If you have failing checks on the configuration overview page, then I would advise fixing those first. There's a big change those mis-configurations are indeed the cause of the failures.

> We can't see any errors on out side.

Do I understand it correctly that there are no tracebacks of this 400 in the container logs?

> In the API specificaties we don't see an API for this.

This is not part of the API nor the API standard, so that would be correct. It's a feature of the Open Zaak admin interface.

> Can you provide us some info (or error) on this?

If you are implying that we check if your environment has any errors, that is not something we can do. Open Zaak does not "phone home" in any way.

**Comment by OlavAllema** [comment]
Thanks Sergei, 
You understand that correctly :) We don't see any tracebacks in the container log.
This is the only log we get (we don't receive any error message or anything):

2022-02-17 13:24:37,323 DEBUG openzaak.utils.middleware  Request headers for /admin/catalogi/zaaktype/28/change/: {'Host': 'dev-openzaak.tezza.eu', 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.80 Safari/537.36', 'Content-Length': '2609', 'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9', 'Accept-Encoding': 'gzip, deflate, br', 'Accept-Language': 'en,nl;q=0.9', 'Cache-Control': 'no-cache', 'Content-Type': 'application/x-www-form-urlencoded', 'Cookie': '_ga=GA1.2.223353551.1590515439; csrftoken=9an6OuwNKiPbAzop47iM7f1oWTOlpJ81NyXAPLrXNkbkStCZGQegI18AhvWs7AC1; openzaak_sessionid=fghiem45h6v9dakfntpte8705bxizb7k', 'Origin': 'https://dev-openzaak.tezza.eu', 'Pragma': 'no-cache', 'Referer': 'https://dev-openzaak.tezza.eu/admin/catalogi/zaaktype/28/change/?_changelist_filters=catalogus__id__exact%3D1', 'Sec-Ch-Ua': '" Not A;Brand";v="99", "Chromium";v="98", "Google Chrome";v="98"', 'Sec-Ch-Ua-Mobile': '?0', 'Sec-Ch-Ua-Platform': '"macOS"', 'Sec-Fetch-Dest': 'document', 'Sec-Fetch-Mode': 'navigate', 'Sec-Fetch-Site': 'same-origin', 'Sec-Fetch-User': '?1', 'Upgrade-Insecure-Requests': '1', 'X-Forwarded-For': '213.127.74.26', 'X-Forwarded-Host': 'dev-openzaak.tezza.eu', 'X-Forwarded-Port': '443', 'X-Forwarded-Proto': 'https', 'X-Forwarded-Server': '5904a8f7d8e0', 'X-Real-Ip': '213.127.74.26'}

[pid: 20|app: 0|req: 1504/3229] 10.0.1.12 () {74 vars in 1754 bytes} [Thu Feb 17 13:24:37 2022] POST /admin/catalogi/zaaktype/28/change/?_changelist_filters=catalogus__id__exact%3D1 => generated 26 bytes in 142 msecs (HTTP/1.1 400) 6 headers in 184 bytes (1 switches on core 1)


Is there something else we can do?

**Comment by sergei-maertens** [comment]
the usual suspect for HTTP 400's is `ALLOWED_HOSTS`, but then you'd also not be able to access the view-config page.

I see you also have #1093, and I'm assuming you've upgraded to see if that is resolved. I also see that #929 might be related. Is there any chance you could attach the export file so we can try to reproduce in a dev environment?

**Comment by OlavAllema** [comment]
Yeah that's right about #1093 . We want to test to export both published and unpublished. For sure, as noticed in #929 we are also expecting the import function to work. 
And to your question, Sergei, how can we export the file when te export function doesn't work? Or do you mean some other 'export file'?

**Comment by sergei-maertens** [comment]
Yeah sorry, I confused things :-) I misread that import doesn't work, but it's export of course.

This is hard to debug/support without more details. Is there any chance you could make a database dump and provide that to us to reproduce, together with any environment variables you may have set for this instance?

The following tables can contain credentials and or PII and can be excluded from the dump:

* `axes_accessattempt`
* `axes_accesslog`
* `drc_cmis_cmisconfig`
* `mozilla_django_oidc_db_openidconnectconfig`
* `notifications_subscription`
* `vng_api_common_apicredential`
* `vng_api_common_jwtsecret`
* `zgw_consumers_service`

**Comment by OlavAllema** [comment]
[openzaak_dump_22-02-2022_15_21_33.zip](https://github.com/open-zaak/open-zaak/files/8132515/openzaak_dump_22-02-2022_15_21_33.zip)


Is this what you need Sergei?

**Comment by sergei-maertens** [comment]
@OlavAllema thank you, this was indeed what I needed and I managed to pinpoint the problem!

When making exports, Open Zaak uses the site configuration to build absolute URLs (URL `/admin/sites/site/`). The domain in that setting happens to go through a validation chain as well where it's checked against the `ALLOWED_HOSTS` setting. This domain is probably missing from your local environment configuration, causing the HTTP 400 error.

I managed to reproduce this in a similar situation, and after editing the site domein in the admin (to `localhost:8000` in my test case), the export works again.

I have created #1106 to follow up on this, but for now you should be able to work around it by using the correct, real domain in the site configuration, or adding the correct service to ALLOWED_HOSTS setting (bearing in mind that the discussion in #1094 is relevant here).

**Comment by sergei-maertens** [comment]
I noticed in your DB dump that the site record in the `django_site` table has primary key `3`. Normally this value is `1` because there's usually only one record. If you have deleted/re-created records in that table, that would lead to auto-increments in the sequence and a different `SITE_ID`, see the relevant docs for this envvar: https://open-zaak.readthedocs.io/en/stable/installation/config/env_config.html#optional

**Comment by TjerkVaags** [comment]
Adding `SITE_ID` did the trick. Thank you for all the help.
